pilot/pkg/model/service.go

```diff
@@ -846,6 +847,43 @@ func (u NoopAmbientIndexes) WorkloadsForWaypoint(scope WaypointScope) []*Workloa
 
 var _ AmbientIndexes = NoopAmbientIndexes{}
 
+type AddressInfo struct {
+	*workloadapi.Address
+}
+
+func (i AddressInfo) ResourceName() string {
+	var name string
+	switch addr := i.Type.(type) {
+	case *workloadapi.Address_Workload:
+		name = GetResourceName(addr.Workload)
+	case *workloadapi.Address_Service:
+		name = GetResourceName(addr.Service)
+	}
+	return name
+}
+
+type ServiceInfo struct {
+	*workloadapi.Service
+}
+
+func (i ServiceInfo) ResourceName() string {
+	return GetResourceName(i.Service)
+}
+
+func GetResourceName(resource interface{}) string {
```

Comment:
nit; rather than have this take any interface, can we have this make a stronger type assertion for common base type such as proto?